subsys/net/lib/lwm2m/lwm2m_obj_connmon.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2019 Foundries.io
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME net_lwm2m_obj_conn_mon
+#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <string.h>
+#include <init.h>
+#include <net/net_if.h>
+#include <net/net_ip.h>
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+
+/* Connectivity Monitoring resource IDs */
+#define CONNMON_NETWORK_BEARER_ID		0
+#define CONNMON_AVAIL_NETWORK_BEARER_ID		1
+#define CONNMON_RADIO_SIGNAL_STRENGTH		2
+#define CONNMON_LINK_QUALITY			3
+#define CONNMON_IP_ADDRESSES			4
+#define CONNMON_ROUTER_IP_ADDRESSES		5
+#define CONNMON_LINK_UTILIZATION		6
+#define CONNMON_APN				7
+#define CONNMON_CELLID				8
+#define CONNMON_SMNC				9
+#define CONNMON_SMCC				10
+
+#define CONNMON_MAX_ID				11
+
+#define CONNMON_STRING_SHORT			8
+
+#if defined(CONFIG_LWM2M_CONN_MON_BEARER_MAX)
```

Comment:
Hmm, is it even possible to have this config undefined, if `LWM2M_CONN_MON_OBJ_SUPPORT` is enabled? If not, I'd remove this check to keep the code cleaner.

Reply:
Yep, it cannot be undefined.  Removing this check.  Thanks.